src/components/input/TypeaheadWithFeedback.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react'
+import Form from 'react-bootstrap/Form'
+
+import { AsyncTypeahead } from 'react-bootstrap-typeahead'
+
+interface Props {
+  id: string
+  searchAccessor: string
+  renderMenuItemChildren: (option: any) => React.ReactNode
+  onChange: (selected: any) => void
+  onSearch: (query: string) => Promise<any[]>
+  minLength?: number
+  placeholder?: string
+  value?: any
+  disabled?: boolean
+  feedback?: string
+  setErrorMessageState?: (arg0: { [key: string]: string } | any) => void
+}
+
+const TypeaheadWithFeebBack = (props: Props) => {
+  const [options, setOptions] = useState<any[]>([])
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const {
+    id,
+    searchAccessor,
+    placeholder,
+    onSearch,
+    onChange,
+    renderMenuItemChildren,
+    minLength,
+    value,
+    disabled,
+    feedback,
+  } = props
+
+  const search = async (query: string) => {
+    setIsLoading(true)
+    const results = await onSearch(query)
+    setOptions(results)
+    setIsLoading(false)
+  }
+
+  const selectedValues = []
+  if (value) {
+    selectedValues.push(value)
+  }
+
+  return feedback ? (
+    <Form.Group>
+      <AsyncTypeahead
+        id={id}
+        labelKey={searchAccessor}
+        options={options}
+        placeholder={placeholder}
+        isLoading={isLoading}
+        minLength={minLength}
+        onSearch={search}
+        onChange={onChange}
+        renderMenuItemChildren={renderMenuItemChildren}
+        defaultSelected={selectedValues}
+        disabled={disabled}
+        isInvalid
+      />
+      <span className="text-danger small">{feedback}</span>
```

Comment:
instead of completely implementing both of the components in each part of the condition, I think we could simplify this conditional render by doing: 

```
 <AsyncTypeahead
      id={id}
      labelKey={searchAccessor}
      options={options}
      placeholder={placeholder}
      isLoading={isLoading}
      minLength={minLength}
      onSearch={search}
      onChange={onChange}
      renderMenuItemChildren={renderMenuItemChildren}
      defaultSelected={selectedValues}
      disabled={disabled}
	  isInvalid={isInvalid}
    />
	{(isInvalid && feedback) && <span className="text-danger small">{feedback}</span>}
```

Reply:
![image](https://user-images.githubusercontent.com/44264255/79033129-a6ec8880-7b79-11ea-8e0a-4a41a21be894.png)

Yes - I saw this prop on the components repo. However, get a TS error when passing this prop.

Reply:
well, this is interesting.. I will investigate quick.

Reply:
I was able to use `feedback` with that component. Try doing a `yarn` command to make sure you have the most recent dependencies.

Reply:
Seemed like running a reinstall on dependencies did the trick